airflow/providers/amazon/aws/hooks/glue_session.py

```diff
@@ -0,0 +1,265 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import asyncio
+import time
+
+from airflow.exceptions import AirflowException
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+
+
+class GlueSessionHook(AwsBaseHook):
+    """
+    Interact with AWS Glue Session.
+
+    Provide thick wrapper around :external+boto3:py:class:`boto3.client("glue") <Glue.Client>`.
+
+    :param id: session id
+    :param desc: session description
+    :param iam_role_name: AWS IAM Role for Glue Session Execution. If set `iam_role_arn` must equal None.
+    :param iam_role_arn: AWS IAM Role ARN for Glue Session Execution, If set `iam_role_name` must equal None.
+    :param num_of_dpus: Number of AWS Glue DPUs to allocate to this Session
+    :param create_session_kwargs: Extra arguments for Glue Session Creation
+
+    Additional arguments (such as ``aws_conn_id``) may be specified and
+    are passed down to the underlying AwsBaseHook.
+
+    .. seealso::
+        - :class:`airflow.providers.amazon.aws.hooks.base_aws.AwsBaseHook`
+    """
+
+    def __init__(
+        self,
+        session_id: str | None = None,
+        desc: str | None = None,
+        iam_role_name: str | None = None,
+        iam_role_arn: str | None = None,
+        num_of_dpus: int | None = None,
+        create_session_kwargs: dict | None = None,
+        session_poll_interval: int | float = 6,
+        *args,
+        **kwargs,
```

Comment:
Are this parameters required into the constructor? And could be this hook are thin wrapper? And provide and validate parameters into the specific hook methods?